command.go

```diff
@@ -896,3 +917,57 @@ func (c *commandContext) SelectSegment(p mat.Vec3) {
 	c.UnsetCursors()
 	c.selectMode = selectModeMask
 }
+
+func (c *commandContext) SelectLabelSegment(p mat.Vec3) error {
+	it, err := c.editor.pp.Vec3Iterator()
+	if err != nil {
+		return err
+	}
+	lt, err := c.editor.pp.Uint32Iterator("label")
+	if err != nil {
+		return err
+	}
+
+	vIndice := make([]int, 0, 8192)
+	n := c.editor.pp.Points
+	for i := 0; i < n; i++ {
+		c.selectMask[i] &= ^uint32(selectBitmaskSegmentSelected)
+		if c.selectMask[i]&(selectBitmaskCropped|selectBitmaskOnScreen) == selectBitmaskOnScreen {
+			v := it.Vec3At(i).Sub(p)
+			if v[0] < -c.labelSegmentationRange || c.labelSegmentationRange < v[0] ||
+				v[1] < -c.labelSegmentationRange || c.labelSegmentationRange < v[1] ||
+				v[2] < -c.labelSegmentationRange || c.labelSegmentationRange < v[2] {
+				continue
+			}
+			vIndice = append(vIndice, i)
+		}
+	}
+
+	raIn := pc.NewIndiceVec3RandomAccessor(it, vIndice)
+	raLIn := pc.NewIndiceUint32RandomAccessor(lt, vIndice)
+
+	searchDistance := c.labelSegmentationSearchDistance
+	kdt := kdtree.New(raIn)
+	nn := kdt.Nearest(p, searchDistance)
+	if nn.ID < 0 {
+		return fmt.Errorf("no point close to %v", p)
+	}
+
+	l := raLIn.Uint32At(nn.ID)
+	if l == 0 {
+		return fmt.Errorf("%v is not labeled", p)
```

Comment:
Isn't there a case user want to select label=0 points?

Reply:
I guess you're right. I'll take that condition out.

Reply:
Done in 9ac3ce813aa2038819c9ae96ec0ac8bbece4ecb6